receiver/singletonreceivercreator/receiver.go

```diff
@@ -0,0 +1,113 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package singletonreceivercreator
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/kyma-project/opentelemetry-collector-components/receiver/singletonreceivercreator/internal/k8sconfig"
+)
+
+var _ receiver.Metrics = (*singletonReceiverCreator)(nil)
+
+// singletonreceivercreator implements consumer.Metrics.
+type singletonReceiverCreator struct {
+	params              receiver.CreateSettings
+	cfg                 *Config
+	nextMetricsConsumer consumer.Metrics
+
+	host              component.Host
+	subReceiverRunner *receiverRunner
+	cancel            context.CancelFunc
+	getK8sClient      func(authType k8sconfig.AuthType) (kubernetes.Interface, error)
+}
+
+func newSingletonReceiverCreator(params receiver.CreateSettings, cfg *Config) component.Component {
+	return &singletonReceiverCreator{
+		params:       params,
+		cfg:          cfg,
+		getK8sClient: k8sconfig.GetK8sClient,
+	}
+}
+
+// Start leader receiver creator.
+func (c *singletonReceiverCreator) Start(_ context.Context, host component.Host) error {
+	c.host = host
+	// Create a new context as specified in the interface documentation
+	ctx := context.Background()
+	ctx, c.cancel = context.WithCancel(ctx)
+
+	c.params.TelemetrySettings.Logger.Debug("Starting leader election receiver...")
```

Comment:
Which log level do other receivers for such messages? Ensure to be consistent. Use consistent naming for the receiver.